### DISCUSSION TITLE
Lots of failed uploads

### DISCUSSION THREAD

**Discussion by vsl-iil** [slskd/slskd#1218] [Q&A] [answered]
I see in my Uploads there are many people that fail to download anything from me. They also pm me and ask why I lock shares, which, in fact, I don't. It's really frustrating to see and it's been like that since I started using slskd. What could be the cause? I have agent-controller setup with my home server sending songs from HDD to the controller over Wireguard VPN. I suspect Internet connection might be to blame, or perhaps it's my controller VPS, which is pretty low spec. Have anyone experienced anything similar?

**Comment by jpdillingham** [comment]
slskd doesn't support locking files, so it's curious that other users would ask about that.

Regarding the failed uploads, it isn't uncommon to see failures here and there but I wouldn't expect it (for most use cases) to be any higher than what you have experienced with Soulseek Qt.

The relay feature, a VPN and a low-spec VPS will all introduce latency and increase the risk of flaky connections.

Can you tell if the transfers are failing to start, or are they starting but failing before they are complete?

**Reply by vsl-iil** [reply]
> Can you tell if the transfers are failing to start, or are they starting but failing before they are complete?

I'm honestly not sure. I tried downloading files from myself with an alt account, but I experienced no problems whatsoever. However, there are many downloads where the first few files seem to download fine, but the rest of an album fails. 

> The relay feature, a VPN and a low-spec VPS will all introduce latency and increase the risk of flaky connections

That's what I think, yes. I'm thinking about hosting the controller on my home server and just investing more time into security — the server itself honestly isn't that critical, but the HDD connected to it is.

**Reply by vsl-iil** [reply, marked as answer]
So I ditched the controller-agent setup over VPN and now just run controller on the home server, and uploads seem to be far more stable, so I'd say that poor connection most definitely was the culprit. I'll try to make a local controller-agent system using docker compose (I suppose it'd be more secure), but for now I'm happy with how it is.
